src/Compiler/SyntaxTree/LexFilter.fs

```diff
@@ -1924,7 +1926,7 @@ type LexFilterImpl (
             returnToken tokenLexbufState token
                 
         //  module ... ~~~> CtxtModuleHead 
-        | MODULE, _ :: _ -> 
+        | MODULE, _ :: _ when offsideStack |> List.forall (fun x -> match x with CtxtParen (LESS _, _) -> false | _ -> true) -> 
```

Comment:
This is the best way to identify that we're within a type application that I could come up with :/